src/Evolution/Systems/Cce/Actions/Psi0Matching.hpp

```diff
@@ -0,0 +1,55 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <tuple>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Evolution/Systems/Cce/NewmanPenrose.hpp"
+#include "Evolution/Systems/Cce/PreSwshDerivatives.hpp"
+
+namespace Cce {
+namespace Actions {
+
+/*!
+ * \ingroup ActionsGroup
+ * \brief Calculate \f$\Psi_0\f$ and its radial derivative
+ * \f$\partial_\underline{\lambda}\Psi_0\f$ at the inner boundary of
+ * CCE grid.
+ *
+ * \details This action is to be called after `Tags::BondiJ` has been updated.
+ * Note that `Tags::BondiJ` is in the asymptotically inertial frame, while the
+ * returned \f$\Psi_0\f$ and its radial derivative are in the Cauchy frame.
+ * \f$\underline \lambda\f$ is an affine parameter along the null
+ * rays generated by \f$l\f$, see Eq. (19a) of \cite Moxon2020gha.
+ */
+struct CalculatePsi0 {
```

Comment:
Is there a better name that better describes what this does?  Maybe `CalculatePsi0AndDerivAtInnerBoundary`?